### PR TITLE
fix generate_short_message and add new spec

### PR DESCRIPTION
**Pull request by bvanmeggelen**
Changes made for https://creativegroupdev.atlassian.net/browse/REC-591.

Allows sending additional fields when using a hash for the message data.

### CODE REVIEW

**Review by rclaassens — APPROVED** [review]

File: lib/rails_graylog/logger.rb [hunk 1]
```diff
@@ -66,7 +66,9 @@ def normalize_message(severity, message)
     end
 
     def generate_short_message(message, writing_object = nil)
-      writing_object.nil? ? "#{message.to_s[0...25]}..." : "#{writing_object.class.name}_#{writing_object.id}"
+      return message[:short_message] if message.is_a?(Hash) && message.key?(:short_message)
+      m = message.is_a?(Hash) && message.key?(:message) ? message[:message] : message
+      writing_object.nil? ? "#{m.to_s[0...25]}..." : "#{writing_object.class.name}_#{writing_object.id}"
```
Comment (bp-behrooz) [inline lib/rails_graylog/logger.rb]: shouldn't we add the `...` only if message is longer than 25 characters?

File: spec/rails_graylog/logger_spec.rb [hunk 2]
```diff
@@ -65,5 +65,19 @@
...
     end
+
+    context 'when logging hash data with message and an additional field' do
+      let(:message_with_field) { { message: Faker::Lorem.sentence, other_field: 'custom value' }}
+
+      it 'calls the graylog notifier with the additional field' do
+        expect(notifier).to receive(:notify!).with(
+          message: message_with_field[:message], full_message: message_with_field[:message],
+          short_message: "#{message_with_field[:message].to_s[0...25]}...",
+          other_field: 'custom value',
```
Comment (bp-behrooz) [inline spec/rails_graylog/logger_spec.rb]: Does this make sense to normalize these hashes? So a bunch of meaningful keys will be in the root and the rest have to be in a different key called `extras`?

File: lib/rails_graylog/logger.rb (hunk 1 above)
Comment (bvanmeggelen) [reply lib/rails_graylog/logger.rb]: Valid point, didn't want to disrupt the current mutations it's doing. I'll take a peek.

**Comment by bvanmeggelen** [comment]
@everplays I updated the logic related to message shortening.

**Review by everplays — APPROVED** [review]
👞 🍑

**Pull request merged**